transaction/core/src/ring_signature/rct_bulletproofs.rs

```diff
@@ -317,13 +305,53 @@ fn sign_with_balance_check<CSPRNG: RngCore + CryptoRng>(
         .map(CompressedCommitment::from)
         .collect();
 
+    // Extend the message with the range proof and pseudo_output_commitments.
+    // This ensures that they are signed.
+    let range_proof_bytes = range_proof.to_bytes();
+    let extended_message = extend_message(message, &pseudo_output_commitments, &range_proof_bytes)?;
+
+    // Prove that the signer is allowed to spend a public key in each ring, and that
+    // the input's value equals the value of the pseudo_output.
+    let mut ring_signatures: Vec<RingMLSAG> = Vec::new();
+    for i in 0..num_inputs {
+        let real_index = real_input_indices[i];
+        let (onetime_private_key, value, blinding) = input_secrets[i];
+        let ring_signature = RingMLSAG::sign(
+            &extended_message,
```

Comment:
This is the main change. Here, the input `message` is the hash of the transaction prefix which does not contain the range proof or pseudo output commitments.